### ISSUE TITLE
How to set sublayer dropout

### ISSUE THREAD

**Issue by egretholier** [lucidrains/x-transformers#318]
attn_dropout and ff_dropout arguments don't add sublayer dropout.

<img width="1313" alt="Image" src="https://github.com/user-attachments/assets/421f6f1d-b0b6-4d3f-86a5-81f4a83a1ab8" />

Sublayer dropout in [The Annotated Transformer](https://nlp.seas.harvard.edu/2018/04/03/attention.html)
```
class SublayerConnection(nn.Module):
    """
    A residual connection followed by a layer norm.
    Note for code simplicity the norm is first as opposed to last.
    """
    def __init__(self, size, dropout):
        super(SublayerConnection, self).__init__()
        self.norm = LayerNorm(size)
        self.dropout = nn.Dropout(dropout)

    def forward(self, x, sublayer):
        "Apply residual connection to any sublayer with the same size."
        return x + self.dropout(sublayer(self.norm(x)))
```

dropout2 in [the implementation of PyTorch TranformerEncoderLayer](https://github.com/pytorch/pytorch/blob/1eba9b3aa3c43f86f4a2c807ac8e12c4a7767340/torch/nn/modules/transformer.py#L948) adds dropout for the ff sublayer.
```
    # feed forward block
    def _ff_block(self, x: Tensor) -> Tensor:
        x = self.linear2(self.dropout(self.activation(self.linear1(x))))
        return self.dropout2(x)
```

**Comment by lucidrains** [comment]
@egretholier hey yea, i could add these, but honestly, they won't make any meaningful difference imo

**Comment by egretholier** [comment]
@lucidrains - Thank you so much for taking the time to respond! I love how you are having a multiplicative impact on the field.

**Comment by egretholier** [comment]
Confirming that sublayer_dropout works.
